### DISCUSSION TITLE
be able to "tail" last N lines of a file

### DISCUSSION THREAD

**Discussion by danielb2** [sharkdp/bat#2243] [Ideas]
so I can use ranges to print the lines, but how would I for example print the last 10 lines of a file? 

I can do this:

`bat file.txt-r $(echo $(wc file.txt | awk  '{print $1}') - 10 | bc):` 

but is there no better way?

**Comment by sharkdp** [comment]
I agree, it would be cool if we could do Python-style negative-number ranges like this:

```bash
bat file.txt -r -10:
```
which would mean: "10 lines from the end until the end". It would be great if you could open a feature request for this.

Until then, your solution seems fine. Another option would be to use `tail` on the input. This might require you to explicitly set the language:
```bash
tail -n10 file.txt | bat -l<language>
```

Or you could use `tail` on the output, but you would need to get rid of the borders (e.g. with `-p`/`--plain`) to make sure that the last line of bats output isn't counted. And you need to force colorization, e.g. with `-f`:
```bash
bat -fp file.txt | tail -n10
```

**Comment by danielb2** [comment]
done here: https://github.com/sharkdp/bat/issues/2251

**Comment by danielb2** [comment]
SO there's no way for bat to know what the last 10 lines of a file are?
